tests/integration/security/egress_sidecar_tls_origination_test.go

```diff
@@ -0,0 +1,203 @@
+				CaCert:      file.AsStringOrFail(t, path.Join(env.IstioSrc, "tests/testdata/certs/dns/root-cert.pem")),
+			}, false, apps.Ns1.Namespace.Name())
+
+			// Set up Host Namespace
+			host := apps.External.All.Config().ClusterLocalFQDN()
+
+			testCases := []struct {
+				name            string
+				statusCode      int
+				credentialToUse string
+				from            echo.Instances
+				drSelector      string
+			}{
+				// Mutual TLS origination from an authorized sidecar to https endpoint
+				{
+					name:            "authorized sidecar",
+					statusCode:      http.StatusOK,
+					credentialToUse: strings.TrimSuffix(credNameGeneric, "-cacert"),
+					from:            apps.Ns1.A,
+					drSelector:      "a",
+				},
+				// Mutual TLS origination from an unauthorized sidecar to https endpoint
+				// This will result in `ERROR Secret is not supplied by SDS`
+				{
+					name:            "unauthorized sidecar",
+					statusCode:      http.StatusBadRequest,
+					credentialToUse: strings.TrimSuffix(credNameGeneric, "-cacert"),
+					from:            apps.Ns1.B,
+					drSelector:      "b",
+				},
+				// Mutual TLS origination using an invalid client certificate
+				// This will result in `400 BadRequest`
+				{
+					name:            "invalid client cert",
+					statusCode:      http.StatusBadRequest,
+					credentialToUse: strings.TrimSuffix(fakeCredName, "-cacert"),
+					from:            apps.Ns1.C,
+					drSelector:      "c",
+				},
+			}
+			for _, tc := range testCases {
+				t.NewSubTest(tc.name).Run(func(t framework.TestContext) {
+					newTLSSidecarDestinationRule(t, apps.External.All, "MUTUAL", tc.drSelector, tc.credentialToUse,
+						apps.Ns1.Namespace)
+					callOpt := newTLSSidecarCallOpts(apps.External.All[0], host, tc.statusCode)
+					tc.from[0].CallOrFail(t, callOpt)
+				})
+			}
+		})
+}
+
+func authorizeSidecar(t framework.TestContext, clientNamespace namespace.Instance, serviceAccountName string) {
+	args := map[string]any{
+		"ServiceAccount": serviceAccountName,
+		"Namespace":      clientNamespace.Name(),
+	}
+
+	role := `
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: allow-list-secrets
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - list
+`
+
+	rolebinding := `
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: allow-list-secrets-to-{{ .ServiceAccount }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: allow-list-secrets
+subjects:
+- kind: ServiceAccount
+  name: {{ .ServiceAccount }}
+  namespace: {{ .Namespace }}
+`
+	t.ConfigIstio().Eval(clientNamespace.Name(), args, role, rolebinding).ApplyOrFail(t, apply.NoCleanup)
+}
+
+func newTLSSidecarDestinationRule(t framework.TestContext, to echo.Instances, destinationRuleMode string,
+	workloadSelector string, credentialName string, clientNamespace namespace.Instance,
```

Comment:
workloadSelector sounds confusing, actually it is the destination service

Reply:
DR API itself has this field as workloadSelector. Doesn't that convey what are the workloads on which this DR should be applied?

Reply:
Normally we call a map workloadSelector